site/content/extend/plugins/webapp/actions.md

```diff
@@ -1,115 +1,493 @@
 
 ```javascript
-import React from 'react';
-import PropTypes from 'prop-types';
+export function getCustomEmoji(emojiId: string): ActionFunc {
+    return bindClientFunc({
+        clientFunc: Client4.getCustomEmoji,
+        onSuccess: EmojiTypes.RECEIVED_CUSTOM_EMOJI,
+        params: [
+            emojiId,
+        ],
+    });
+}
+```
 
-export default class ActionExample extends React.PureComponent {
-    static propTypes = {
-        user: PropTypes.object.isRequired,
-        patchUser: PropTypes.func.isRequired, // here we define the action as a prop
-    }
+**createPost**
+
+Dispatch this action to create a new post.
 
-    updateFirstName = () => {
-        const patchedUser = {
-            id: this.props.user.id,
-            first_name: 'Jim',
+```javascript
+export function createPost(post: Post, files: any[] = []) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState();
+        const currentUserId = state.entities.users.currentUserId;
+
+        const timestamp = Date.now();
+        const pendingPostId = post.pending_post_id || `${currentUserId}:${timestamp}`;
+        let actions: Action[] = [];
+
+        if (Selectors.isPostIdSending(state, pendingPostId)) {
+            return {data: true};
+        }
+
+        let newPost = {
+            ...post,
+            pending_post_id: pendingPostId,
+            create_at: timestamp,
+            update_at: timestamp,
+            reply_count: 0,
         };
 
-        this.props.patchUser(patchedUser); // here we use the action
-    }
+        if (post.root_id) {
+            newPost.reply_count = Selectors.getPostRepliesCount(state, post.root_id) + 1;
+        }
+
+        // We are retrying a pending post that had files
+        if (newPost.file_ids && !files.length) {
+            files = newPost.file_ids.map((id) => state.entities.files.files[id]); // eslint-disable-line
+        }
+
+        if (files.length) {
+            const fileIds = files.map((file) => file.id);
+
+            newPost = {
+                ...newPost,
+                file_ids: fileIds,
+            };
+
+            actions.push({
+                type: FileTypes.RECEIVED_FILES_FOR_POST,
+                postId: pendingPostId,
+                data: files,
+            });
+        }
+
+        actions.push({
+            type: PostTypes.RECEIVED_NEW_POST,
+            data: {
+                ...newPost,
+                id: pendingPostId,
+            },
+        });
 
-    render() {
-        return (
-            <div>
-                {'First name: ' + this.props.user.first_name}
-                <a
-                    href='#'
-                    onClick={this.updateFirstName}
-                >
-                    Click me to update the first name!
-                </a>
-            </div>
-        );
-    }
+```
+
+**getMyTeams**
+
+Dispatch this action to fetch all the team types associated with.
```

Comment:
Associated with what/whom?

Reply:
Associated with the current user. The changes have been made.